src/__test__/components/SearchPanel.test.tsx

```diff
@@ -33,61 +27,63 @@ describe('SearchPanel component', () => {
       isLoading: false,
     };
 
-    window.localStorage.setItem('tg-last-search', JSON.stringify(mockState));
+    const getItemSpy = vi
+      .spyOn(Storage.prototype, 'getItem')
+      .mockReturnValue(JSON.stringify(mockState));
 
     render(<SearchPanel />);
 
+    expect(getItemSpy).toHaveBeenCalledWith('tg-last-search');
     expect(screen.getByDisplayValue('test')).toBeInTheDocument();
+
+    getItemSpy.mockRestore();
   });
 
   it('loadPokemon: loads data with query', async () => {
     render(<SearchPanel />);
 
-    const mockFetch = () => {
-      return vi.fn().mockResolvedValue({
-        ok: true,
-        status: 200,
-        json: async () => mockPokemon,
-      });
-    };
-
-    globalThis.fetch = mockFetch();
+    const mockFetch = vi.spyOn(window, 'fetch').mockResolvedValue({
+      ok: true,
+      status: 200,
+      json: async () => mockPokemon,
+    } as Response);
 
     await userEvent.type(screen.getByRole('textbox'), 'pokemon');
     await userEvent.click(screen.getByRole('button', { name: /Search/ }));
 
     await waitFor(() => {
       const cells = screen.getAllByRole('cell', { name: /pokemon/i });
       expect(cells.length).toBeGreaterThan(0);
+      expect(mockFetch).toBeCalled();
     });
+
+    mockFetch.mockRestore();
   });
 
   it('loadPokemon: loads data with error', async () => {
     const consoleError = vi
       .spyOn(console, 'error')
       .mockImplementation(() => {});
 
-    render(<SearchPanel />);
-
-    const mockFetch = () => {
-      return vi.fn().mockResolvedValue({
-        ok: true,
-        status: 200,
-        json: async () => {
-          throw new Error('fetch error');
-        },
-      });
-    };
+    const mockFetch = vi.spyOn(window, 'fetch').mockResolvedValue({
```

Comment:
it is better to use same approach in another places where you use `globalThis.fetch = mockFetch()` and clean fetch mock in the en of the test